pybamm/experiment/step/base_step.py

```diff
@@ -100,8 +102,11 @@ def __init__(
                     f"Input function must return a real number output at t = {t0}"
                 )
 
+        # Record whether the step uses the default duration
+        # This will be used by the experiment to check whether the step is feasible
+        self.uses_default_duration = duration is None
```

Comment:
I assume a zero duration should also trigger the default, not just if one is not passed in, correct?

Reply:
Zero duration should error, I'll add a separate check for that